Framework/Core/include/Framework/ASoA.h

```diff
@@ -1763,9 +1773,9 @@ template <typename T>
 using is_soa_filtered_t = typename framework::is_base_of_template<soa::FilteredPolicy, T>;
 
 template <typename T>
-auto filter(T&& t, framework::expressions::Filter const& expr)
+auto filter(T const& t, framework::expressions::Filter&& f)
 {
-  return Filtered<T>(t.asArrowTable(), expr);
+  return t.select(f);
```

Comment:
I think the other way around would be better: define the object oriented code in terms of the functional one. Also I would rename it consistently either both select or both filter.

Reply:
Understood, I will move the functionality to an external function and use other tutorial example.